test/io/test_cli.py

```diff
@@ -289,3 +290,75 @@ def test_jwt_aud_config_set_to_invalid_uri(defaultenv):
     with pytest.raises(PostgrestError):
         dump = cli(["--dump-config"], env=env).split("\n")
         assert "jwt-aud should be a string or a valid URI" in dump
+
+
+def test_cli_ready_flag_success(defaultenv):
+    "PostgREST ready flag succeeds when ready"
+
+    env = {
+        **defaultenv,
+        "PGRST_ADMIN_SERVER_PORT": "3001",
+    }
+
+    # Run postgrest process
+    pgrst_process = run_postgrest_binary(env)
+
+    time.sleep(3)  # Wait so postgrest gets ready
+    try:
+        # Run healthcheck process
+        command = [POSTGREST_BIN] + ["--ready"]
+        healthcheck_process = subprocess.run(command, env=env, capture_output=True)
+        assert healthcheck_process.returncode == 0
+        assert f"OK: http://:3001/ready" in healthcheck_process.stdout.decode()
+    finally:
+        kill_pgrst_process(pgrst_process)
+
+
+def test_cli_ready_flag_fail(defaultenv):
+    "PostgREST ready flag fails when not ready"
+
+    env = {
+        **defaultenv,
+        "PGRST_ADMIN_SERVER_PORT": "3001",
+    }
+
+    # Run postgrest process
+    pgrst_process = run_postgrest_binary(env)
+
+    time.sleep(0)  # No waiting and instantly run the healthcheck process.
+    try:
+        # Run healthcheck process
+        command = [POSTGREST_BIN] + ["--ready"]
+        healthcheck_process = subprocess.run(command, env=env, capture_output=True)
+        assert healthcheck_process.returncode == 1
+        assert (
+            "postgrest: health check request failed - connection refused to"
+            in healthcheck_process.stderr.decode()
+        )
```

Comment:
This test is flaky and fails sometimes because the `subprocess.run` exits without capturing the stderr output. I guess this happens due to output not being flushed properly. Any ideas on how to handle this cleanly?

Reply:
Maybe instead try to force a schema cache failure so `/ready` always fails. An idea could be to put a bad database connection uri. Check the io tests too, likely this was done before.